src/modules/xml/producer_xml.c

```diff
@@ -1147,7 +1147,8 @@ static void on_end_consumer( deserialise_context context, const xmlChar *name )
 			{
 				// Instantiate the consumer
 				char *id = trim( mlt_properties_get( properties, "mlt_service" ) );
-				context->consumer = mlt_factory_consumer( context->profile, id, resource );
+				mlt_profile consumer_profile = mlt_profile_clone( context->profile );
```

Comment:
This looks pretty good except we are leaking the cloned profile here. It might be safe to store as a data property with destructor on the consumer. I will test that.

Reply:
Here is my change on top, and it is working OK (not segfaulting) for me.